services/web/server/src/simcore_service_webserver/utils_aiohttp.py

```diff
@@ -9,9 +17,56 @@ def rename_routes_as_handler_function(routes: RouteTableDef, *, prefix: str):
         route.kwargs["name"] = f"{prefix}.{route.handler.__name__}"
 
 
-def view_routes(routes: RouteTableDef):
+def get_routes_view(routes: RouteTableDef) -> str:
     fh = io.StringIO()
     print(routes, file=fh)
     for r in routes:
         print(" ", r, file=fh)
     return fh.getvalue()
+
+
+def create_url_for_function(request: web.Request) -> Callable:
+    app = request.app
+
+    def url_for(router_name: str, **params) -> Optional[str]:
+        try:
+            rel_url: URL = app.router[router_name].url_for(
+                **{k: f"{v}" for k, v in params.items()}
+            )
+            url = (
+                request.url.origin()
+                .with_scheme(
+                    request.headers.get("X-Forwarded-Proto", request.url.scheme)
+                )
+                .with_path(str(rel_url))
+            )
+            return f"{url}"
+
+        except KeyError:
+            return None
```

Comment:
question, what is the use case to return None?

Reply:
Actually on second thought, it is safer to raise. Added an exception with a comprehensive message